### DISCUSSION TITLE
1주차 Discussion

### DISCUSSION THREAD

**Discussion by chloe-codes1** [slippStudy/java-network-study#1] [General]
## 📢 책을 읽으면서 궁금한 점들을 자유롭게 적고 대답해 주세요

1주차 진행 방식

> 진행자: 정규님

- 진도: 1, 2장 읽고, 질문 준비해 오기

**Comment by LenKIM** [comment]
궁금한거
- 실무에서 네트워크 지식이 유용했던 적이 있는가?
- 멀티 스레드는 언제 사용한다고 생각하고 있는가?

**Comment by jinwookh** [comment]
궁금한 점
- 실무에서 어느 네트워크 계층까지 알아야 유용할까?
- 2장 맨 마지막 예시를 보면 consumer가 producer보다 먼저 start하여, producer가 시작할 때까지 wait하도록 하고 있다. 실무에서 쓰레드를 wait시켜본 경험이 있나요? 혹은 특정 라이브러리에서 사용된 것을 본 적 있나요?

**Reply by jinwookh** [reply]
- Layer 4 정도까지 알면 되지 않을까?
- wait을 실무에 사용한 경험은 없다. ConcurrentHashMap에서 많이 사용된 것을 본 적 있다.

**Comment by chloe-codes1** [comment]
### 궁금한 점

1. Multi-thread를 구현할 때 주로 어떤 방식을 사용하고, 선호하나요?
    - java.util.concurrent
    - CompletableFuture
    - ForkJoin
    - …
2. Java 21의 Virtual Thread를 사용해 본 적이 있나요? 기존 multi-thread model의 문제점을 해결해서 사용한 경험이 있나요?

**Comment by kangbada0728** [comment]
1. OSI 7계층은 제대로 지켜지고 있는가?
2. java.util.concurrent  를 배우려면 어떤것을 구현해보면 좋을까?

**Comment by yangyunhyeuk** [comment]
1. 개발하며 OSI 7 계층을 염두하며 개발했던 경험이 있는가?
2. 개발하며 스레드 우선순위를 컨트롤했던 경험이 있는가?
